Adding ability to systematically modify yaml files from ansible.

kwoodson opened the pull request:
This pull request adds the ability to systematically modify yaml|json files through ansible.

@maxamillion

Commit:
Adding ability to systematically modify yaml files from ansible.

gundalow commented on lib/ansible/modules/files/yedit.py:
```diff
@@ -0,0 +1,965 @@
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'core'}
```
community, no core here please.

kwoodson replied:
I was copying what another file had in it.  I'll update.

gundalow commented on lib/ansible/modules/files/yedit.py:
```diff
@@ -0,0 +1,965 @@
+
+DOCUMENTATION = '''
+---
+module: yedit
```
Bike shed: would be good if this included `yaml` in its name.

kwoodson replied:
yedit was chosen for yaml - edit.  It also handles json since it is a subset of yaml.  Basically modifying structured yaml/json.

gundalow commented on lib/ansible/modules/files/yedit.py:
```diff
@@ -0,0 +1,965 @@
+    pass
+
+
+# pylint: disable=too-many-public-methods,too-many-instance-attributes
```
`RETURN` block missing

kwoodson replied:
I'm not sure what this returns to.  I'll look at some other modules to determine what this refers to.

kwoodson replied:
Updated.

kwoodson commented:
@gundalow, Thanks.   Addressed the comments.  The bot states that it cannot find yaml.  How do we handle this failure?  I am using PyYaml or ruamel.yaml.

kwoodson commented:
@gundalow updated with your requested comments.

kwoodson commented:
@gundalow, any updates on this?

maxamillion commented:
ping @gundalow

ssbarnea commented:
Can we revive this, is a feature I was dreaming about for years!

marksteward commented:
Is anybody working on this? Is there any functionality missing? If this stands a chance of being merged I'm happy to fix any bugs/compatibility issues.
